File: 0110-balanced-binary-tree/0110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True
        return (self.height(root) >= 1)
    def height(self, root):
        if not root:
            return 0
        else:
            left_height = self.height(root.left)
            right_height = self.height(root.right)
            if left_height < 0 or right_height < 0 or abs(left_height - right_height) > 1:
                return -1
            return max(left_height, right_height) + 1

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.dart
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *   int val;
 *   TreeNode? left;
 *   TreeNode? right;
 *   TreeNode([this.val = 0, this.left, this.right]);
 * }
 */
import 'dart:math';
class Solution {
  bool isBalanced(TreeNode? root) {
    if (root == null) {
        return true;
    }
    if (cal(root).runtimeType == bool) {
        return false;
    } else {
        return true;
    }
  }
    dynamic cal(TreeNode? root) {
        if (root == null) {
            return 0;
        }
        var temp_left = cal(root.left);
        if (temp_left.runtimeType == bool) {
            return false;
        }
        var temp_right = cal(root.right);
        if (temp_right.runtimeType == bool) {
            return false;
        }
        print((temp_left - temp_right).abs());
        if ((temp_left - temp_right).abs() > 1) {
            return false;
        }
        if (temp_right > temp_left) {
            return temp_right + 1;
        } else {
            return temp_left + 1;
        }
    }
}